src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Windows.cs

```diff
@@ -750,5 +752,25 @@ internal unsafe void FreeNativeOverlapped(NativeOverlapped* pOverlapped)
                 }
             }
         }
+
+        // Abstracts away the type signature incompatibility between Memory and ReadOnlyMemory.
+        // TODO: Use abstract static methods when they become stable.
```

Comment:
Why are we doing this complicated dance with generics rather than just treating everything as `Memory<byte>` via MemoryMarshal.AsMemory?

Reply:
I see, we're trying to unify arrays of these types.

Reply:
We can indeed unify `Memory` and `ReadOnlyMemory` but not _`IReadOnlyList`s_ of them. This trick provides a common implementation and minimizes code duplication.